### PR TITLE
ref: upgrade linters to flake8 5.x

### PR DESCRIPTION
**Pull request by asottile-sentry**
noqa removal was automated via [yesqa](https://github.com/asottile/yesqa)

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -1,16 +1,17 @@
 [flake8]
-ignore =
```
Comment (asottile-sentry) [inline .flake8:2]: extend-ignore is better than ignore, it preserves the default ignore

File: .flake8 [hunk 2]
```diff
@@ -1,16 +1,17 @@
 [flake8]
-ignore = 
-  E203,  // Handled by black (Whitespace before ':' -- handled by black)
-  E266,  // Handled by black (Too many leading '#' for block comment)
-  E501,  // Handled by black (Line too long)
-  W503,  // Handled by black (Line break occured before a binary operator)
-  E402,  // Sometimes not possible due to execution order (Module level import is not at top of file)
-  E731,  // I don't care (Do not assign a lambda expression, use a def)
```
Comment (asottile-sentry) [inline .flake8:8]: ini comments are either `#` or `;` not `//` -- these are actually being interpreted as ignore codes https://flake8.pycqa.org/en/latest/user/configuration.html

File: .flake8 [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
-  B950,  // Handled by black (Line too long by flake8-bugbear)
-  B011,  // I don't care (Do not call assert False)
-  B014,  // does not apply to Python 2 (redundant exception types by flake8-bugbear)
-  N812,  // I don't care (Lowercase imported as non-lowercase by pep8-naming)
-  N804   // is a worse version of and conflicts with B902 (first argument of a classmethod should be named cls)
-max-line-length = 80
-select = N,B,C,E,F,W,T4,B9
```
Comment (asottile-sentry) [inline .flake8:15]: this isn't necessary -- flake8 enables the codes of all extension by default

File: .flake8 [hunk 4]
```diff
@@ -1,16 +1,17 @@
...
-exclude=checkouts,lol*,.tox
```
Comment (asottile-sentry) [inline .flake8:16]: extend-exclude is better as it preserves the default exclude list

File: sentry_sdk/api.py [hunk 5]
```diff
@@ -130,7 +130,7 @@ def configure_scope(  # noqa: F811
 
 
 @overload
-def push_scope():  # noqa: F811
```
Comment (asottile-sentry) [inline sentry_sdk/api.py:133]: I fixed pyflakes a while back to understand `@overload`

File: linter-requirements.txt [hunk 6]
```diff
@@ -1,10 +1,9 @@
 black==22.3.0
-flake8==3.9.2
-flake8-import-order==0.18.1
```
Comment (asottile-sentry) [inline linter-requirements.txt:3]: this wasn't actually being executed (since it wasn't part of the `select` list below) so I removed it

**Comment by antonpirker** [comment]
Thanks for this update @asottile-sentry !
I will review this after DjangoCon

**Review by antonpirker — APPROVED** [review]

File: linter-requirements.txt (hunk 6 above)
Comment (antonpirker) [reply linter-requirements.txt:3]: ok

File: sentry_sdk/api.py (hunk 5 above)
Comment (antonpirker) [reply sentry_sdk/api.py:133]: 👍

**Pull request merged**